### ISSUE TITLE
Word segmentation is incorrect

### ISSUE THREAD

**Issue by robertbastian** [unicode-org/icu4x#5015]

**Comment by sffc** [comment]
@eggrobin What is left on this issue?

**Comment by eggrobin** [comment]
> What is left on this issue?

All of it? It was created to allow us to close the specific issue reported in https://github.com/unicode-org/icu4x/issues/4417, but word segmentation is still wrong and hasn’t changed since this was filed.

**Comment by makotokato** [comment]
I thought that this was fixed when I added random tests in all UAX#29 segmenter in 2.0 development cycles.

**Comment by eggrobin** [comment]
Yes, I see https://github.com/unicode-org/icu4x/pull/6442 fixed this (by duplicating most of the states rather than hoisting the ZWJ handling into the code).
